### PR TITLE
refine tid parsing logic

### PR DESCRIPTION
**Pull request by Nesze**

### CODE REVIEW

File: kafka-bridge.go [hunk 1]
```diff
@@ -137,18 +138,23 @@ func extractJSON(msg string) (jsonContent string, err error) {
 	return jsonContent, err
 }
 
-func extractTID(msg string) (tid string, err error) {
-	if !strings.Contains(msg, "X-Request-Id") {
-		return tid, errors.New("X-Request-Id header could not be found.")
+func extractTID(msg string) (string, error) {
+	header := regexp.MustCompile("X-Request-Id:.*").FindString(msg)
```
Comment (martingartonft) [inline kafka-bridge.go]: move this to a package variable to avoid compiling the regex every time.

File: kafka-bridge.go [hunk 2]
```diff
@@ -137,18 +138,23 @@ func extractJSON(msg string) (jsonContent string, err error) {
...
+	if tid == "" {
+		return "", fmt.Errorf("Transaction ID is in unknown format: %s.", header)
 	}
-	startIndex := strings.Index(msg, "X-Request-Id: tid_") + len("X-Request-Id: ")
-	tid = msg[startIndex : startIndex+len("tid_")+10]
 	return tid, nil
 }
 
+func findTID(header string) string {
+	tidRegexp := regexp.MustCompile("(tid|SYN-REQ-MON)[a-zA-Z0-9_]*$")
```
Comment (martingartonft) [inline kafka-bridge.go]: same here.

File: kafka-bridge.go [hunk 3]
```diff
@@ -137,18 +138,23 @@ func extractJSON(msg string) (jsonContent string, err error) {
...
+	return tidRegexp.FindString(header)
```
Comment (martingartonft) [inline kafka-bridge.go]: after moving the regex into a variable, this is just one line, so inline  it?

File: kafka-bridge.go (hunk 1 above)
Comment (Nesze) [inline kafka-bridge.go]: I've missed this again, thanks!

**Pull request merged**